File: backend/src/com/origin/entity/InventoryItem.java
package com.origin.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

/**
 * предмет в инвентаре
 */
@Entity
@Table(name = "inventory")
public class InventoryItem
{
	@Id
	@Column(name = "id", columnDefinition = "INT(11) NOT NULL AUTO_INCREMENT")
	private int _id;

	/**
	 * ид инвентаря (родителя, вещи в которой находится этот предмет
	 */
	@Column(name = "inventoryId", columnDefinition = "INT(11) NOT NULL")
	private int _inventoryId;

	/**
	 * тип предмета
	 */
	@Column(name = "type", columnDefinition = "INT(11) NOT NULL")
	private int _type;

	/**
	 * положение внутри инвентаря
	 */
	@Column(name = "x", columnDefinition = "INT(11) NOT NULL")
	private int _x;

	@Column(name = "y", columnDefinition = "INT(11) NOT NULL")
	private int _y;

	/**
	 * качество вещи
	 */
	@Column(name = "quality", columnDefinition = "INT(11) NOT NULL")
	private int _quality;

	/**
	 * количество в стаке
	 */
	@Column(name = "count", columnDefinition = "INT(11) NOT NULL")
	private int _count;

	/**
	 * тик (если вещь может имзенятся с течением времени
	 */
	@Column(name = "tick", columnDefinition = "INT(11) NOT NULL")
	private int _tick;

	public int getId()
	{
		return _id;
	}

	public int getInventoryId()
	{
		return _inventoryId;
	}

	public void setInventoryId(int inventoryId)
	{
		_inventoryId = inventoryId;
	}

	public int getType()
	{
		return _type;
	}

	public void setType(int type)
	{
		_type = type;
	}

	public int getX()
	{
		return _x;
	}

	public void setX(int x)
	{
		_x = x;
	}

	public int getY()
	{
		return _y;
	}

	public void setY(int y)
	{
		_y = y;
	}

	public int getQuality()
	{
		return _quality;
	}

	public void setQuality(int quality)
	{
		_quality = quality;
	}

	public int getCount()
	{
		return _count;
	}

	public void setCount(int count)
	{
		_count = count;
	}

	public int getTick()
	{
		return _tick;
	}

	public void setTick(int tick)
	{
		_tick = tick;
	}
}

File: backend/src/com/origin/entity/InventoryItem.kt
package com.origin.entity

import org.jetbrains.exposed.dao.LongEntity
import org.jetbrains.exposed.dao.LongEntityClass
import org.jetbrains.exposed.dao.id.EntityID
import org.jetbrains.exposed.dao.id.LongIdTable

/**
 * предмет в инвентаре
 */
object InventoryItems : LongIdTable("inventory") {
    /**
     * ид инвентаря (родителя, вещи в которой находится этот предмет
     */
    val inventoryId = integer("inventoryId")

    /**
     * тип предмета
     */
    val type = integer("type")

    /**
     * положение внутри инвентаря
     */
    val x = integer("x")
    val y = integer("y")

    /**
     * качество вещи
     */
    val quality = integer("quality")

    /**
     * количество в стаке
     */
    val count = integer("count").default(1)

    /**
     * тик (если вещь может имзенятся с течением времени
     */
    val tick = integer("tick").default(0)
}

class InventoryItem(id: EntityID<Long>) : LongEntity(id) {
    companion object : LongEntityClass<InventoryItem>(InventoryItems)

    var inventoryId by InventoryItems.inventoryId
    var type by InventoryItems.type
    var x by InventoryItems.x
    var y by InventoryItems.y
    var quality by InventoryItems.quality
    var count by InventoryItems.count
    var tick by InventoryItems.tick
}